sympy/geometry/point.py

```diff
@@ -266,8 +266,14 @@ def distance(self, p):
         sqrt(x**2 + y**2)
 
         """
-        p = Point(p)
-        return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
+        try:
+            args = p.args
+        except AttributeError:
+            try:
+                args = [as_int(i) for i in p]
```

Comment:
Actually one `as_int` is defined in `from sympy.core.compatibility import as_int`. I used that recently in `Group Theory` so i remember. Perhaps that is not available publicily.